agent/repowise.py

```
@@ -0,0 +1,283 @@
+                    deps.append(f"  - callee: {imp}")
+            except Exception:
+                pass
+
+        if "callers" in include:
+            try:
+                module_name = path.stem
+                if "__init__" in module_name:
+                    module_name = path.parent.name
+
+                cmd = f"grep -lE '(import|from).*\b{module_name}\b' -r . --exclude-dir={{.git,__pycache__,.venv,node_modules}}"
+                result = subprocess.run(cmd, shell=True, cwd=self.root, capture_output=True, text=True)
```

Ruff (0.15.12)

[error] `subprocess` call with `shell=True` identified, security issue

(S602)